File: backend/AkilliSatici.Api/AkilliSatici.Api/Models/AppUser.cs
using System.ComponentModel.DataAnnotations;

namespace AkilliSatici.Api.Models;

public class AppUser
{
    public int Id { get; set; }

    [Required, MaxLength(120)]
    public string FullName { get; set; } = "";

    [Required, MaxLength(120)]
    public string StoreName { get; set; } = "";

    [Required, MaxLength(200)]
    [EmailAddress]
    public string Email { get; set; } = "";

    public bool IsAdmin { get; set; }

    [Required]
    public byte[] PasswordHash { get; set; } = Array.Empty<byte>();

    [Required]
    public byte[] PasswordSalt { get; set; } = Array.Empty<byte>();

    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

    public int PawPoints { get; set; } = 0;

    public List<Product> Products { get; set; } = new();

    public List<PetProfile> PetProfiles { get; set; } = new();

    public string? ResetCode { get; set; }

    public DateTime? ResetCodeExpires { get; set; }
}

File: web/lib/src/models/app_user.dart
class AppUser {
  final int id;
  final String fullName;
  final String email;
  final String storeName;
  final bool isAdmin;
  final int pawPoints;

  const AppUser({
    required this.id,
    required this.fullName,
    required this.email,
    required this.storeName,
    required this.isAdmin,
    this.pawPoints = 0,
  });

  factory AppUser.fromJson(Map<String, dynamic> json) {
    return AppUser(
      id: json['id'] as int,
      fullName: (json['fullName'] ?? '') as String,
      email: (json['email'] ?? '') as String,
      storeName: (json['storeName'] ?? '') as String,
      isAdmin: (json['isAdmin'] ?? false) as bool,
      pawPoints: (json['pawPoints'] ?? 0) as int,
    );
  }
}
